consensus/state.go

```diff
@@ -756,17 +756,33 @@ func (cs *ConsensusState) handleTimeout(ti timeoutInfo, rs cstypes.RoundState) {
 func (cs *ConsensusState) handleTxsAvailable() {
 	cs.mtx.Lock()
 	defer cs.mtx.Unlock()
-	// we only need to do this for round 0
-	cs.enterNewRound(cs.Height, 0)
-	cs.enterPropose(cs.Height, 0)
+
+	// We only need to do this for round 0.
+	if cs.Round != 0 {
+		return
+	}
+
+	switch cs.Step {
+	case cstypes.RoundStepNewHeight: // timeoutCommit phase
+		if cs.needProofBlock(cs.Height) {
```

Comment:
don't like that we're loading block meta here, but I guess it's fine for now

Reply:
`needProofBlock` to avoid race between this func and `enterNewRound` https://github.com/tendermint/tendermint/blob/master/consensus/state.go#L835

`SkipTimeoutCommit` I've realized that we don't need it here since 

a) if we're in timeoutCommit phase, it means we haven't received all the votes (https://github.com/tendermint/tendermint/blob/master/consensus/state.go#L1574) and it's ok to wait 
b) if we're in propose phase, we don't need to check SkipTimeoutCommit and can `enterPropose` as we do right now